controlplane/kubeadm/internal/etcd_client_generator.go

```diff
@@ -56,3 +58,25 @@ func (c *etcdClientGenerator) forNode(ctx context.Context, name string) (*etcd.C
 	}
 	return customClient, nil
 }
+
+// forLeader takes a list of nodes and returns a client to the leader node
+func (c *etcdClientGenerator) forLeader(ctx context.Context, nodes *corev1.NodeList) (*etcd.Client, error) {
+	for _, node := range nodes.Items {
```

Comment:
Before this for loop, let's create a new aggregate error message like `var errs []error`